diffkemp/simpll/DebugInfo.cpp

```diff
@@ -86,31 +86,32 @@ void DebugInfo::extractAlignmentFromInstructions(GetElementPtrInst *GEP,
         for (auto idx = GEP->idx_begin();
                 idx != GEP->idx_end();
                 ++idx, indices.push_back(*idx)) {
-            auto indexedType =
-                    GEP->getIndexedType(GEP->getSourceElementType(),
+            auto indexedType = GEP->getIndexedType(GEP->getSourceElementType(),
                                         ArrayRef<Value *>(
                                                 indices));
 
             Type *indexedTypeOther = nullptr;
             if (OtherGEP)
-                    indexedTypeOther = OtherGEP->getIndexedType(
+                indexedTypeOther = OtherGEP->getIndexedType(
                                         OtherGEP->getSourceElementType(),
                                         ArrayRef<Value *>(
-                                                indicesOther));
+                                            indicesOther));
 
-            if (!indexedType->isStructTy()) {
-                indices.push_back(*idx);
+            if (!indexedType->isStructTy())
                 continue;
-            }
 
-            if (indexedTypeOther && idx_other != OtherGEP->idx_end() &&
-                !indexedTypeOther->isStructTy()) {
-                indices.push_back(*idx_other);
-                continue;
+            if (indexedTypeOther && !indexedTypeOther->isStructTy()) {
+                // The pushing of idx_other is done at the end of the cycle, not
+                // in the for header, therefore it has to be pushed here
+                if (idx_other != OtherGEP->idx_end())
+                    indicesOther.push_back(*idx_other);
+
```

Comment:
I think that we should either 'continue' here or do not insert *idx_other into indicesOther since this will be done in the end of the loop again.